### PR TITLE
[stdlib] implement os.path.split

### PR DESCRIPTION
**Pull request by artemiogr97**
Implementation of `os.path.split`

### CODE REVIEW

File: stdlib/test/os/path/test_split.mojo [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    var head: String
+    var tail: String
+    head, tail = split("/path/to/file")
+    assert_equal(head, "/path/to")
+    assert_equal(tail, "file")
+    head, tail = split(Path("/path/to/file"))
+    assert_equal(head, "/path/to")
+    assert_equal(tail, "file")
+
+    head, tail = split("/path/to/file/")
```
Comment (rparolin) [inline stdlib/test/os/path/test_split.mojo:32]: I believe this is a typo. "/path/to/file/" should "/path/to/directory/"

File: stdlib/test/os/path/test_split.mojo [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    assert_equal(head, "")
+    assert_equal(tail, "file")
+    head, tail = split(Path("file"))
+    assert_equal(head, "")
+    assert_equal(tail, "file")
+
+    head, tail = split("///")
+    assert_equal(head, "///")
+    assert_equal(tail, "")
+    head, tail = split(Path("///"))
```
Comment (rparolin) [inline stdlib/test/os/path/test_split.mojo:84]: Why is the same test copy/pasted?

**Review by rparolin — APPROVED** [review]
Looks good.  Thank you for the contribution.  Please add a PR description and I'll merge this PR internally.

**Comment by rparolin** [comment]
!sync

File: stdlib/test/os/path/test_split.mojo (hunk 2 above)
Comment (artemiogr97) [reply stdlib/test/os/path/test_split.mojo:84]: you mean everywhere? It is to test both overloads

**Comment by artemiogr97** [comment]
@rparolin I see that this PR was left open, should I do a rebase to solve the conflicts and be able to close it after that?

**Comment by JoeLoser** [comment]
@rparolin do you plan on still marshaling this through?

**Comment by JoeLoser** [comment]
Closing this PR as this is already implemented: see https://github.com/modularml/mojo/blob/61d8208e3beedb4ec4a35b73c91cc9169fa46591/stdlib/src/os/path/path.mojo#L344

**Pull request closed without merging**